### PR TITLE
feat: form page

### PR DESCRIPTION
**Pull request by csiyang**
# Description

### <samp>🤖 Generated by Copilot at 37c9497</samp>

This pull request adds a new feature to the journeys-admin app, which is a dynamic onboarding form that uses the Formium service. It also adds a new FormiumForm component to the shared-ui library, which renders the Formium form using custom MUI components. The pull request also includes a story file to demonstrate the FormiumForm component in Storybook.

[- Create page](https://3.basecamp.com/3105655/buckets/34356604/todos/6557492664)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] should be able to see and submit a form on `/onboarding-form`

# Walkthrough

### <samp>🤖 Generated by Copilot at 37c9497</samp>

*  Add a new page component for the onboarding form using Formium and next-firebase-auth ([link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-719483df49549a751a6b58ff274ddf6312157c1665900db7e79e213f88feb30eR1-R59))
*  Create and export a Formium client instance using environment variables ([link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-6da455094dff3e0be6001d235afa69f27d81f9637988828f42f1f715ec273b83R1-R8), [link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-3d37d91516cb3664d853a265e7272b4f15838159b50b8368e73ba3400e456857R1))
*  Render the Formium form using custom components and a PageWrapper component ([link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-e2e419b179c8247666bfff842d0c96b4e94d397225b6830fc57c2193c972112eR1-R58), [link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-f70c9b2f784e3e2dec58f2ed74aba4e6523a7b347df46d4d5bb8d36cb2a1e25bR1-R12), [link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-6d62fa24702afa29e362f9f76ec0adc2e0506fc946a409aa76b75c48f0b5969bR1))
*  Define custom components for each field element using MUI components and a FormControl component ([link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-18979a638ae0c71c25aa8bdfccc021a69a6d0b874b6dcd8e8c85800f40c8ed52R1-R13), [link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-f1275b417fd22eac7fb4c15086e46d57e979bd202ab198ac29005dadca3760f2R1), [link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-9efbffe3bb0781a5b733a39e8e795735103422dcbd3b140dc71d408aac3ba18aR1-R32), [link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-de2524e26016501287a36c94654726fa8eb8ecffeab08fa9b00f140bb6f0312cR1))
*  Wrap the field elements with a Stack component for spacing and margin ([link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-9f431ca7bb4865ab9f4434b43a6d0a119c4397a2f86f7dbaef7865948ff433ecR1-R16), [link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-03c9899b93b8c3e206a2746ca8d08c08d45882db309edf1e1948f82fea2dde4bR1))
*  Render the header for each page of the form using the form data and a Box component ([link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-1372929933c5118e1d5334f60b860d2494f1c41ca55204e9866074c8d1815c77R1-R54), [link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-9acba841e0bc7a24f143e3715bb92263a2f07acd83256b7d8393affef6a9e2d4R1-R2))
*  Render the next and previous buttons for the form using the button element props and a Button component ([link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-92d1b98a10c30020c08df45121a32f5f8a6329f4927430a30d78fabfafe25754R1-R14), [link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-0af347309bba6c96f3493ad2f2112dfad2fb1f75169376935f4a61e9a5295827R1), [link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-af382441d182e263240decb21e21006742fdf7b822624387679529656c33339bR1-R14))
*  Wrap the footer component with a Box component for centering ([link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-15d51310ed7bb4b87cdf85443645434e6ec1d51fca540b0ceaf746a1605a75abR1-R14), [link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-05867510bd3aad16368e6d1ed7ed96ba56e7a618b09465b5031e69c62ba17a0fR1))
*  Add a story file to demonstrate the FormiumForm component and its subcomponents using Storybook and sharedUiConfig ([link](https://github.com/JesusFilm/core/pull/1888/files?diff=unified&w=0#diff-ce054a956c4bdcac15b7462e7088a49e4a0017757de209f0d9566727e38e5cdfR1-R162))

### CODE REVIEW

File: libs/shared/ui/src/components/FormiumForm/Header/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { Form, FormElementAction, FormElementType } from '@formium/types'
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import { ReactElement } from 'react'
+
+/** Contains information for the page and renders the header */
+
+export interface HeaderProps {
```
Comment (csiyang) [inline libs/shared/ui/src/components/FormiumForm/Header/Header.tsx:8]: This is typed as `any` within formium.

I really only needed to pull the title out of all this, but decided to just type everything based off what the console log was coming back with

File: apps/journeys-admin/pages/onboarding-form/index.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+function OnboardingFormPage({ form }: Props): ReactElement {
+  return (
+    <Box
+      sx={{
+        display: 'flex',
+        justifyContent: 'center'
+      }}
+    >
+      <FormiumForm formiumClient={formiumClient} form={form} />
```
Comment (tataihono) [inline apps/journeys-admin/pages/onboarding-form/index.tsx:27]: can the formium client be moved to shared UI?

File: apps/journeys-admin/pages/onboarding-form/index.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  withAuthUserTokenSSR
+} from 'next-firebase-auth'
+import { ReactElement } from 'react'
+
+import { FormiumForm } from '@core/shared/ui/FormiumForm'
+
+import { formiumClient } from '../../src/libs/formiumClient'
+import { initAndAuthApp } from '../../src/libs/initAndAuthApp'
+
+interface Props {
```
Comment (tataihono) [inline apps/journeys-admin/pages/onboarding-form/index.tsx:15]:
```suggestion
interface OnboardingFormPageProps {
```

File: libs/shared/ui/src/components/FormiumForm/ElementsWrapper/ElementsWrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import Stack from '@mui/material/Stack'
+import { ReactElement } from 'react'
+
+/** Wraps around the all of the field elements */
```
Comment (tataihono) [inline libs/shared/ui/src/components/FormiumForm/ElementsWrapper/ElementsWrapper.tsx:4]: Is this passed as a jsdoc comment with a line underneath it?

File: libs/shared/ui/src/components/FormiumForm/ElementsWrapper/ElementsWrapper.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+export function ElementsWrapper({
+  children
+}: {
+  children: ReactElement
+}): ReactElement {
```
Comment (tataihono) [inline libs/shared/ui/src/components/FormiumForm/ElementsWrapper/ElementsWrapper.tsx:10]: to remain consistent please extract all these props out as interfaces.

File: libs/shared/ui/src/components/FormiumForm/FormiumForm.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  SubmitButton,
+  NextButton,
+  PreviousButton
+}
+
+interface Props {
+  formiumClient: FormiumClient
+  form: Form
+  // userId?: string
+  // email?: string
```
[the remark this answers is not in the document]
Comment (tataihono) [reply libs/shared/ui/src/components/FormiumForm/FormiumForm.tsx:44]: drop the comment

File: libs/shared/ui/src/components/FormiumForm/FormiumForm.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+export function FormiumForm({ formiumClient, form }: Props): ReactElement {
+  return (
+    <Formium
+      data={form}
+      components={myComponents}
+      onSubmit={async (values) => {
+        await formiumClient.submitForm('ns-test', values)
+        alert('Success')
```
Comment (tataihono) [inline libs/shared/ui/src/components/FormiumForm/FormiumForm.tsx:54]: no alert

**Review by tataihono — CHANGES_REQUESTED** [review]
this still needs to be integrated into the onboarding flow. it should be handled in this PR or the page should be cut from this pr if you are trying to ship the shared UI component separately.

**Comment by csiyang** [comment]
requires 
- https://github.com/JesusFilm/core/pull/1933
- https://github.com/JesusFilm/core/pull/1931

File: libs/shared/ui/src/libs/FormiumProvider/FormiumProvider.tsx
Comment (tataihono) [inline libs/shared/ui/src/libs/FormiumProvider/FormiumProvider.tsx:1]: merge into formium form.

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/libs/checkConditionalRedirect/checkConditionalRedirect.ts [hunk 8]
```diff
@@ -40,10 +42,16 @@ export async function checkConditionalRedirect(
       destination: `/users/terms-and-conditions${redirect}`,
       permanent: false
     }
+  } else if (
+    (flags.onboardingForm === true || flags.onboardingForm == null) &&
```
Comment (csiyang) [inline apps/journeys-admin/src/libs/checkConditionalRedirect/checkConditionalRedirect.ts:46]: update flags check

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — APPROVED** [review]

File: libs/shared/ui/src/components/FormiumForm/Header/Header.tsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  return (
+    <Box
+      data-testid="Header"
+      sx={{
+        display: 'flex',
+        justifyContent: 'center',
+        pb: '30px'
+      }}
+    >
+      <Stack spacing={2}>
```
Comment (csiyang) [inline libs/shared/ui/src/components/FormiumForm/Header/Header.tsx:51]: Needs updating

**Review by tataihono — APPROVED** [review]

**Review by tataihono — APPROVED** [review]

**Pull request merged**